Sources/NIO/SelectableEventLoop.swift

```diff
@@ -202,7 +202,7 @@ internal final class SelectableEventLoop: EventLoop {
             self._tasksLock.withLockVoid {
                 self._scheduledTasks.remove(task)
             }
-            self._wakeupSelector()
+            // we don't need to wake up the selector here, the scheduled task will never be picked up.
```

Comment:
The wake up here was to potentially recalculate the timeout. It might be nice to clarify that in this comment: even if this was the next task we were going to process, we’ll wake up once unnecessarily which isn’t too expensive, whereas waking up the selector every time is very expensive.

Reply:
great idea! done.